src/ssl.c

```diff
@@ -16307,10 +16972,77 @@ int wolfSSL_set_compression(WOLFSSL* ssl)
             ret = WOLFSSL_SUCCESS;
         }
 
+    #ifdef WOLFSSL_DUAL_ALG_CERTS
+        if (ret == WOLFSSL_SUCCESS)
+            /* Set the label for the alternative key, too. User can still
+             * override that afterwards. */
+            ret = wolfSSL_use_AltPrivateKey_Label(ssl, label, devId);
+    #endif
+
         return ret;
     }
 #endif /* WOLF_PRIVATE_KEY_ID */
 
+#if defined(WOLF_PRIVATE_KEY_ID) && defined(WOLFSSL_DUAL_ALG_CERTS)
+    int wolfSSL_use_AltPrivateKey_id(WOLFSSL* ssl, const unsigned char* id,
+                                  long sz, int devId, long keySz)
+    {
+        int ret = wolfSSL_use_AltPrivateKey_Id(ssl, id, sz, devId);
+
+        if (ret == WOLFSSL_SUCCESS)
+            ssl->buffers.altKeySz = (word32)keySz;
+
+        return ret;
+    }
+
+    int wolfSSL_use_AltPrivateKey_Id(WOLFSSL* ssl, const unsigned char* id,
+                                     long sz, int devId)
+    {
+        int ret = WOLFSSL_FAILURE;
+
+        if (ssl->buffers.weOwnAltKey)
+            FreeDer(&ssl->buffers.altKey);
+        if (AllocDer(&ssl->buffers.altKey, (word32)sz, ALT_PRIVATEKEY_TYPE,
+                                                            ssl->heap) == 0) {
+            XMEMCPY(ssl->buffers.altKey->buffer, id, sz);
+            ssl->buffers.weOwnAltKey = 1;
+            ssl->buffers.altKeyId = 1;
+            if (devId != INVALID_DEVID)
+                ssl->buffers.altKeyDevId = devId;
+            else
+                ssl->buffers.altKeyDevId = ssl->devId;
+
+            ret = WOLFSSL_SUCCESS;
+        }
+
+        return ret;
+    }
+
+    int wolfSSL_use_AltPrivateKey_Label(WOLFSSL* ssl, const char* label,
+                                        int devId)
+    {
+        int ret = WOLFSSL_FAILURE;
+        word32 sz = (word32)XSTRLEN(label) + 1;
+
```

Comment:
Sanity checks on arguments passed in here for public facing function.

Reply:
Fixed in next commit.